DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_priority.py

```diff
@@ -0,0 +1,98 @@
+    6. Verify traffic on receiving ports.
+    7. Verify priority was not stripped from the packet.
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 2)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    vlan_priority = 6
+    vlan_id = 5
+
+    port_map = ({"port": 0, "settings": [{"vlan": vlan_id, "untagged": False, "pvid": True}]},
+                {"port": 1, "settings": [{"vlan": vlan_id, "untagged": False, "pvid": True}]})
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3. Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"Traffic with VLAN priority": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "srcMac": "02:00:00:00:00:01",
+        "dstMac": "02:00:00:00:00:02",
+        "vlanID": vlan_id,
+        "vlanPriority": vlan_priority,
+        "egress_track_by": ["vlan", "pcp"]
+    }}
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(5)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # 6. Verify traffic.
+    # 7. Verify priority was not stripped from the packet.
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+        egress_stats = await tgen_utils_get_egress_stats(tgen_dev, row)
+        for idx, ege_row in enumerate(egress_stats.Rows):
+            if idx == 0:
+                continue
+            assert int(ege_row["Egress Tracking 2"]) == vlan_priority, "No/Wrong VLAN priority in the packet"
```

Comment:
Do you check that amount of packets with this priority is an expected one?

Reply:
No, only presence of the priority tag

Reply:
If I'm not mistaken, you can check the count of packets with the specific priority. Please do so if it is possible.